browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -302,6 +344,36 @@ void SpeedreaderTabHelper::DidStopLoading() {
   }
 }
 
+void SpeedreaderTabHelper::DOMContentLoaded(
+    content::RenderFrameHost* render_frame_host) {
+  if (!PageWantsDistill(distill_state_))
+    return;
+
+  constexpr int kIsolatedWorldId = content::ISOLATED_WORLD_ID_CONTENT_END + 1;
+
+  constexpr const char16_t kAddShowOriginalPageLink[] =
+      uR"js(
+    (function() {
+      const link = document.getElementById('show-original-page');
+      if (!link)
+        return;
+      link.text = '$1';
```

Comment:
there should be some sanitization here; otherwise if a translator could do a JS injection by setting `$1` to be `'; someJavaScript(); '`. i believe there are some scripts which check that our translations don't contain unsafe HTML but it wouldn't catch javascript injection.